lib/hammer_cli_foreman_puppet/commands.rb

```diff
@@ -0,0 +1,55 @@
+module HammerCLIForemanPuppet
+  RESOURCE_NAME_MAPPING = {
+    :puppetclass => :puppet_class,
+    :puppetclasses => :puppet_classes,
+    :environment => :puppet_environment,
+    :environments => :puppet_environments
+  }.freeze
+
+  RESOURCE_ALIAS_NAME_MAPPING = {
+    environment: :puppet_environment,
+    environments: :puppet_environments
+  }.freeze
+
+  module ResolverCommons
+    def self.included(base)
+      base.extend(ClassMethods)
+    end
+    module ClassMethods
+      def resolver
+        HammerCLIForemanPuppet::IdResolver.new(
+          HammerCLIForemanPuppet.api_connection,
+          HammerCLIForemanPuppet::Searchables.new
+        )
+      end
+      def resource_name_mapping
+        HammerCLIForemanPuppet::RESOURCE_NAME_MAPPING.dup
+      end
+
+      def searchables
+        @searchables ||= HammerCLIForemanPuppet::Searchables.new
+        @searchables
+      end
+    end
+  end
+
```

Comment:
I guess it's would be safer to add also
```ruby
class Command < HammerCLIForeman::Command
  include HammerCLIForemanPuppet::ResolverCommons
end
```
and then re-use where needed instead of `HammerCLIForeman::Command`. Katello does such thing.